compiler/catala_utils/message.ml

```diff
@@ -350,6 +349,18 @@ open Content
 exception CompilerError of Content.t
 exception CompilerErrors of Content.t list
 
+type error_kind = Lexing | Parsing | Typing | Generic
+
+type error = {
+  kind : error_kind;
+  msg : string;
+  pos : Pos.t option;
+  suggestion : string list option;
+}
```

Comment:
What's the use for this? Doesn't it duplicate `Content.t` ?

Reply:
It mostly does, I was wary of exposing `Content.t` but if you think that's fine, we can do it. Perhaps as a private type.